File: lib/app/data/models/shopping_item.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'item_unit.dart';

part 'shopping_item.freezed.dart';
part 'shopping_item.g.dart';

enum ShoppingStatus {
  pending,
  purchased;

  String get displayName {
    switch (this) {
      case ShoppingStatus.pending:
        return 'Pending';
      case ShoppingStatus.purchased:
        return 'Purchased';
    }
  }
}

@freezed
class ShoppingItem with _$ShoppingItem {
  const factory ShoppingItem({
    required String id,
    required String name,
    @Default('') String category,
    @Default(1) int quantity,
    @Default(1) int priority, // 1-5
    @Default(ShoppingStatus.pending) ShoppingStatus status,
    String? linkedQuestId,
    // Unit tracking
    @Default(ItemUnit()) ItemUnit unit, // Unit type (e.g., "kg", "bottles", "packs")
  }) = _ShoppingItem;

  factory ShoppingItem.fromJson(Map<String, dynamic> json) =>
      _$ShoppingItemFromJson(json);
}

File: backend/app/models/shopping_item.py
"""ShoppingItem model matching Flutter's shopping_item.dart."""

from datetime import datetime
from enum import Enum
from pydantic import BaseModel, Field

from .common import ItemUnit


class ShoppingStatus(str, Enum):
    """Shopping item status."""
    pending = "pending"
    purchased = "purchased"


class ShoppingItemBase(BaseModel):
    """Base shopping item fields."""
    name: str = Field(..., min_length=1, max_length=200)
    category: str = ""
    quantity: int = 1
    priority: int = Field(default=1, ge=1, le=5)
    status: ShoppingStatus = ShoppingStatus.pending
    linked_quest_id: str | None = None
    unit: ItemUnit = Field(default_factory=ItemUnit)


class ShoppingItemCreate(ShoppingItemBase):
    """Shopping item creation request."""
    pass


class ShoppingItemUpdate(BaseModel):
    """Shopping item update request."""
    name: str | None = Field(None, min_length=1, max_length=200)
    category: str | None = None
    quantity: int | None = None
    priority: int | None = Field(None, ge=1, le=5)
    status: ShoppingStatus | None = None
    linked_quest_id: str | None = None
    unit: ItemUnit | None = None


class ShoppingItem(ShoppingItemBase):
    """Shopping item response model."""
    id: str
    user_id: str
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True
        use_enum_values = True

